dot/types/block_data.go

```diff
@@ -28,7 +28,7 @@ func NewEmptyBlockData() *BlockData {
 
 // Number returns the block header number.
 func (bd *BlockData) Number() uint {
-	return bd.Header.Number
+	return uint(bd.Header.Number)
```

Comment:
no need right? It's already a `uint`

```suggestion
	return bd.Header.Number
```